clients/src/main/java/org/apache/kafka/clients/ClientTelemetryProvider.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients;
+
+import io.opentelemetry.proto.common.v1.AnyValue;
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.proto.resource.v1.Resource;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.Configurable;
+import org.apache.kafka.common.metrics.MetricsContext;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ClientTelemetryProvider implements Configurable {
+
+    public static final String DOMAIN = "org.apache.kafka";
+    // Client metrics tags
+    public static final String CLIENT_RACK = "client_rack";
+    public static final String GROUP_ID = "group_id";
+    public static final String GROUP_INSTANCE_ID = "group_instance_id";
+    public static final String GROUP_MEMBER_ID = "group_member_id";
+    public static final String TRANSACTIONAL_ID = "transactional_id";
+
+    private static final String PRODUCER_NAMESPACE = "kafka.producer";
+    private static final String CONSUMER_NAMESPACE = "kafka.consumer";
+
+    private static final Map<String, String> PRODUCER_CONFIG_MAPPING = new HashMap<>();
+    private static final Map<String, String> CONSUMER_CONFIG_MAPPING = new HashMap<>();
+
+    private volatile Resource resource = null;
+    private Map<String, ?> config = null;
+
+    static {
+        PRODUCER_CONFIG_MAPPING.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, ClientTelemetryProvider.TRANSACTIONAL_ID);
+
+        CONSUMER_CONFIG_MAPPING.put(ConsumerConfig.GROUP_ID_CONFIG, ClientTelemetryProvider.GROUP_ID);
+        CONSUMER_CONFIG_MAPPING.put(ConsumerConfig.GROUP_INSTANCE_ID_CONFIG, ClientTelemetryProvider.GROUP_INSTANCE_ID);
```

Comment:
It seem we should add `group.member.id` there, too?

Reply:
`group.member.id` is not required to be added in this static mapping, as the properly is not defined in config rather arrives in response from join request. Once the response arrives the `updateMetricsLabel` is called to update resource metrics labels.